tests/Testcontainers.Tests/Unit/Builders/DockerConfigTest.cs

```diff
@@ -0,0 +1,39 @@
+namespace DotNet.Testcontainers.Tests.Unit
+{
+  using System;
+  using System.Diagnostics;
+  using DotNet.Testcontainers.Builders;
+  using Xunit;
+
+  public class DockerConfigTest
+  {
+    [Fact]
+    public void GetCurrentEndpoint()
+    {
+      var endpoint = new DockerConfig().GetCurrentEndpoint();
+      Assert.NotNull(endpoint);
+
+      var expectedEndpoint = DockerProcess.GetCurrentEndpoint();
+      Assert.Equal(expectedEndpoint, endpoint);
+    }
+  }
+
+  internal static class DockerProcess
+  {
+    public static Uri GetCurrentEndpoint()
+    {
+      using var docker = new Process();
+      docker.StartInfo = new ProcessStartInfo
+      {
+        FileName = "docker",
+        Arguments = "context inspect --format {{.Endpoints.docker.Host}}",
+        RedirectStandardOutput = true,
+        UseShellExecute = false,
+      };
+      docker.Start();
+      docker.WaitForExit(2000);
+      var endpoint = docker.StandardOutput.ReadToEnd().Trim();
+      return new Uri(endpoint);
+    }
```

Comment:
Probably we can reuse the [DockerCLI](https://github.com/testcontainers/testcontainers-dotnet/blob/87184e3e53b270ac352cdf59af6c2334488c7fd1/tests/Testcontainers.Commons/DockerCli.cs#L44-L54) class here (gettin the host value).

Reply:
Absolutely! Done in 512784ffcb43f257757039f6c31d5230899666e1.